### PR TITLE
Proper Invalid Bankswitch Emulation

### PR DESCRIPTION
**Pull request by ParzivalWolfram**
I've added proper invalid-bank switching behavior. Should make some homebrew and bootlegs more stable.
(Don't mind my stupid commit names)

### CODE REVIEW

File: platform/ds/arm7/source/main.c [hunk 1]
```diff
@@ -145,9 +145,6 @@ void powerHandler(u32 value, void* user_data) {
             }
             fifoSendValue32(FIFO_SYSTEM, battery);
             break;
-        case PM_DSI_HACK:
-            __dsimode = true;
-            break;
```
Comment (Drenn1) [inline platform/ds/arm7/source/main.c]: Maybe libnds has evolved, but can I ask why you had to remove this?

File: platform/ds/arm9/Makefile [hunk 2]
```diff
@@ -31,7 +31,6 @@ DEBUG	:=	-g
 CFLAGS	:=	-Wall -Wno-psabi -DNO_VIZ \
  			-march=armv5te -mtune=arm946e-s -fomit-frame-pointer\
 			-ffast-math -O3 $(ARCH) $(DEBUG) \
-			-DVERSION_STRING=\"`git describe --always --abbrev=4`\"\
```
Comment (Drenn1) [inline platform/ds/arm9/Makefile]: I guess this didn't work for you? I'd really prefer to keep this though...

Comment (Drenn1) [reply platform/ds/arm9/Makefile]: Nevermind, I see you added it back later...

**Comment by Drenn1** [comment]
Also, so I can test this later, can you give an example of a homebrew that this fixes?

**Comment by Drenn1** [comment]
To be fair, it's been a while since I compiled this branch. Maybe some of those changes were needed. (I more recently worked on the v0.5-hotfix branch.)

**Comment by ParzivalWolfram** [comment]
Yeah, there are several issues with the pre-existing code... of course, my
added code caused even more, which is why I undid my changes.

**Comment by ParzivalWolfram** [comment]
`\GameYob/platform/ds/arm7/source/main.c: In function 'powerHandler':
GameYob/platform/ds/arm7/source/main.c:148:14: error: 'PM_DSI_HACK' undeclared (first use in this function)
         case PM_DSI_HACK:
              ^~~~~~~~~~~`
Yeah, it's busted. xD

**Comment by ParzivalWolfram** [comment]
Fixed the tab issue.

**Comment by ParzivalWolfram** [comment]
@Drenn1 I fixed the tab issue (i think)

**Comment by ParzivalWolfram** [comment]
yeah no, just reject this

i'm just gonna make it a parallel project

**Comment by Drenn1** [comment]
Well, feel free to make another PR when this is ready... no reason to fork the codebase unless this has a large speed impact.

**Pull request closed without merging**